src/panels/config/devices/ha-config-device-page.ts

```diff
@@ -145,8 +145,15 @@ export class HaConfigDevicePage extends LitElement {
   );
 
   private _integrations = memoizeOne(
-    (device: DeviceRegistryEntry, entries: ConfigEntry[]): ConfigEntry[] =>
-      entries.filter((entry) => device.config_entries.includes(entry.entry_id))
+    (device: DeviceRegistryEntry, entries: ConfigEntry[]): ConfigEntry[] => {
+      const entryLookup: { [entryId: string]: ConfigEntry } = {};
+      for (const entry of entries) {
+        entryLookup[entry.entry_id] = entry;
+      }
+      return device.config_entries
+        .map((entry) => entryLookup[entry])
+        .filter(Boolean);
```

Comment:
Why the filter? Can the device have an config entry id that no longer exists?

Reply:
config entries registry can be empty when page load.